packages/cli-helpers/__mocks__/fs.js

```diff
@@ -1,220 +1,4 @@
-import path from 'path'
+import * as memfs from 'memfs'
```

Comment:
I know Josh and I talked about mocking the filesystem, and the automocking stuff.

So just wanted to confirm this is the way you guys prefer to do this. Is it?
(I'm a little worried I said something that pushed you towards this solution, but that wasn't my intention, if I did)

Reply:
I think I prefer going with an externally provided mock, like memfs. Shifting the responsibility onto a third party to match feature parity with fs seems like something we'll be really glad to do. It's already possible to see minor changes needed, like now we have to supply 'utf-8' to readFileSync as our previous mocking didn't quite match the real implementation which by default returns a buffer and not a string. 

I think the specific implementation of the auto mock (`__mock__/fs.js`) can still be improved for sure. There are cases in this PR where we still use specific mocks of fs within the test files rather than rely on the automock. My thinking is that we'll have a series of test refactoring PRs once we've switched over and have had the experience/learning we need to know how best to get things like automocking done.

Do you have a preference or a different approach in mind? This is certainly not set in stone or anything.

Reply:
I just wanted to make sure you didn't (keep) using the auto mock feature because of something I said if you didn't want to. Because right now I don't have a preference one way or the other

Reply:
Okay perfect thanks!